nilearn/reporting/html_report.py

```diff
@@ -0,0 +1,341 @@
+                          docstring=docstring,
+                          parameters=parameters,
+                          description=description)
+
+    head_template_name = 'report_head_template.html'
+    head_template_path = resource_path.joinpath(head_template_name)
+    with open(str(head_template_path), 'r') as head_file:
+        head_tpl = Template(head_file.read())
+
+    return HTMLReport(body=body, head_tpl=head_tpl)
+
+
+def _remove_after_n_seconds(file_name, n_seconds):
+    script = os.path.join(os.path.dirname(__file__), 'rm_file.py')
+    subprocess.Popen(['python', script, file_name, str(n_seconds)])
+
+
+class ReportMixin:
+    """
+    A class to provide general reporting functionality
+    """
+
+    def _define_overlay(self):
+        """
+        Determine whether an overlay was provided and
+        update the report text as appropriate.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        """
+        displays = self._reporting()
+
+        if len(displays) == 1:  # set overlay to None
+            overlay, image = None, displays[0]
+
+        elif len(displays) == 2:
+            overlay, image = displays[0], displays[1]
+
+        return overlay, image
+
+    def generate_report(self):
+        """
+        Generate a report for Nilearn objects.
+
+        Report is useful to visualize steps in a processing pipeline.
+        Example use case: visualize the overlap of a mask and reference image
+        in NiftiMasker.
+
+        Returns
+        -------
+        report : HTMLReport
+        """
+        if not hasattr(self, 'input_'):
+            warnings.warn('Report generation not enabled !'
+                          'No visual outputs will be created.')
+            report = _update_template(title='Empty Report',
+                                      docstring=('This report was not '
+                                                 'generated. Please check '
+                                                 'that reporting is enabled.'),
+                                      content=_embed_img(None),
+                                      overlay=_embed_img(None),
+                                      parameters=dict())
+
+        else:
+            overlay, image = self._define_overlay()
+            description = self._report_description
+            parameters = _str_params(self.get_params())
+            docstring = self.__doc__
+            snippet = docstring.partition('Parameters\n    ----------\n')[0]
+            report = _update_template(title=self.__class__.__name__,
+                                      docstring=snippet,
+                                      content=_embed_img(image),
+                                      overlay=_embed_img(overlay),
+                                      parameters=parameters,
+                                      description=description)
+        return report
+
+
+class HTMLDocument(object):
```

Comment:
maybe this could go in its own module in the `reporting` package? that's a matter of taste so feel free to ignore

Reply:
I think that might be good. It seems to be a versatile class, also used by my GLM reporter in Nistats, which will move into Nilearn soon-ish. Might be cleaner to have it in its own module. Up to you EMdupre.

Reply:
Done in https://github.com/nilearn/nilearn/pull/2019/commits/4aa81d33cec67b63767e6190174736073e4c78b0 !

Reply:
thanks!